eth_utils/address.py

```diff
@@ -119,65 +118,72 @@ def _normalize_32byte_address(address):
         raise ValueError("Invalid address.  Must be 32 byte value")
 
 
-@coerce_args_to_text
-@coerce_return_to_text
-def to_normalized_address(address):
+def to_normalized_address(primitive=None, hexstr=None, text=None):
     """
     Converts an address to it's normalized hexidecimal representation.
     """
-    if is_hex_address(address):
+    address = to_text(primitive, hexstr, text)
+    if is_hex_address(text=address):
         return _normalize_hex_address(address)
-    elif is_binary_address(address):
+    elif is_binary_address(text=address):
         return _normalize_binary_address(address)
-    elif is_32byte_address(address):
+    elif is_32byte_address(text=address):
         return _normalize_32byte_address(address)
 
     raise ValueError("Unknown address format {0}".format(address))
 
 
-def is_normalized_address(value):
+def is_normalized_address(primitive=None, hexstr=None, text=None):
     """
     Returns whether the provided value is an address in it's normalized form.
     """
-    if not is_address(value):
+    if not is_address(primitive, hexstr, text):
         return False
     else:
-        return value == to_normalized_address(value)
+        value = next(item for item in [primitive, hexstr, text] if item is not None)
+        return value == to_normalized_address(primitive, hexstr, text)
 
 
-@coerce_args_to_bytes
-@coerce_return_to_bytes
-def to_canonical_address(address):
+def to_canonical_address(primitive=None, hexstr=None, text=None):
     """
+    Given any supported representation of an address
+    returns it's canonical form (20 byte long string).
     """
-    return decode_hex(to_normalized_address(address))
+    byte_address = to_bytes(primitive, hexstr, text)
+    return decode_hex(to_normalized_address(byte_address))
 
 
-def is_canonical_address(value):
-    if not is_address(value):
+def is_canonical_address(primitive=None, hexstr=None, text=None):
+    """
+    Returns `True` if the `value` is an address in it's canonical form.
+    """
+    if not is_address(primitive, hexstr, text):
         return False
     else:
-        return value == to_canonical_address(value)
+        value = next(item for item in [primitive, hexstr, text] if item is not None)
```

Comment:
Is there a simpler way to pull out the non-None arg?

Reply:
`any([primitive, hexstr, text])` would do it

... but I think the nature of this function is that you don't need to split the argument into three. Either it is the canonical representation or it's not. We don't need to test the canonical-ness of bytes, it's clearly not. As for what to do with calling `is_address`, I'm not sure that a `text` representation of an address ever really gets used in practice. But if we want to keep the three-way split in `is_address()` anyway, for consistency, it seems totally reasonable here to do:

```py
def is_canonical_address(address):
    if not is_string(address) or not is_hex(address):
        return False
    elif not is_address(hexstr=address):
        return False
    else:
        return address == to_canonical_address(hexstr=address)
```